src/components/History/History.jsx

```diff
@@ -1,91 +1,96 @@
 import { useNavigate } from "react-router-dom";
-import { useState } from "react";
 import { SideBar } from "../SideBar/SideBar";
 import { useData } from "../../contexts";
-import { addWatchLater, deleteHistory } from "../../services";
+import { useUserActions } from "../../hooks";
+import { PlaylistModal } from "../PlaylistModal/PlaylistModal";
+import { SaveToPlaylist } from "../ActionItems/SaveToPlaylist";
+import { WatchLaterActions } from "../ActionItems/WatchLaterActions";
+import { Loader } from "../Loader/Loader";
 
 export const History = () => {
-  const { data, dispatch } = useData();
-  const [id, setId] = useState();
+  const { data, id, setId, loading, loadtext } = useData();
   const navigate = useNavigate();
-  const token = localStorage.getItem("login");
+  const { deleteHistoryVideo, clearHistoryVideos } = useUserActions();
 
   const showSingleVideo = (video) => {
     navigate(`/singlevideo/${video._id}`);
   };
 
-  const deleteHisoryVideo = async (video) => {
-    setId(0);
-    const historyResponse = await deleteHistory({
-      videoId: video._id,
-      encodedToken: token,
-    });
-    dispatch({ type: "LOAD_HISTORY", payload: historyResponse.data.history });
-  };
-
-  const addWatchlater = async (video) => {
-    setId(0);
-    const watchRes = await addWatchLater({ video: video, encodedToken: token });
-    dispatch({ type: "LOAD_WATCHLATER", payload: watchRes.data.watchlater });
-  };
-
   return (
     <div className="grid-container">
       <div className="sidebar">
         <SideBar />
       </div>
-      <div className="content video-list">
-        {data.history.map((video) => {
-          return (
-            <div className="video-card" key={video._id}>
-              <div
-                className="video-image"
-                onClick={() => showSingleVideo(video)}
+      <div className="content">
+        {loading ? (
+          <Loader text={loadtext} />
+        ) : (
+          <>
+            <div className="section-header">
+              <h2 className="playlist-name">History</h2>
+              <button
+                className="btn btn-solid-primary"
+                onClick={() => clearHistoryVideos()}
               >
-                <img
-                  src={video.videoThumbnail}
-                  alt="thumb"
-                  className="img-responsive"
-                />
-              </div>
+                Clear History
+              </button>
+            </div>
+            <div className="video-list">
+              {data.history.map((video) => {
+                return (
+                  <div className="video-card" key={video._id}>
+                    <div
```

Comment:
Suggestion: you could remove curly braces and the return keyword as you are returning only a single line of code, i.e., `<div></div>`